homeassistant/components/bayesian/config_flow.py

```diff
@@ -0,0 +1,526 @@
+"""Config flow for the Bayesian integration."""
+
+from collections.abc import Mapping
+from enum import StrEnum
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    DOMAIN as BINARY_SENSOR_DOMAIN,
+    BinarySensorDeviceClass,
+)
+from homeassistant.components.input_boolean import DOMAIN as INPUT_BOLEAN_DOMAIN
+from homeassistant.components.input_number import DOMAIN as INPUT_NUMBER_DOMAIN
+from homeassistant.components.number import DOMAIN as NUMBER_DOMAIN
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import (
+    CONF_ABOVE,
+    CONF_BELOW,
+    CONF_DEVICE_CLASS,
+    CONF_ENTITY_ID,
+    CONF_NAME,
+    CONF_STATE,
+    CONF_TYPE,
+    CONF_VALUE_TEMPLATE,
+)
+from homeassistant.helpers import selector
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaCommonFlowHandler,
+    SchemaConfigFlowHandler,
+    SchemaFlowError,
+    SchemaFlowFormStep,
+    SchemaFlowMenuStep,
+)
+
+from .binary_sensor import above_greater_than_below, no_overlapping
+from .const import (
+    CONF_INDEX,
+    CONF_OBSERVATIONS,
+    CONF_P_GIVEN_F,
+    CONF_P_GIVEN_T,
+    CONF_PRIOR,
+    CONF_PROBABILITY_THRESHOLD,
+    CONF_TEMPLATE,
+    CONF_TO_STATE,
+    DEFAULT_NAME,
+    DEFAULT_PROBABILITY_THRESHOLD,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ObservationTypes(StrEnum):
+    """StrEnum for all the different observation types."""
+
+    STATE = CONF_STATE
+    NUMERIC_STATE = "numeric_state"
+    TEMPLATE = CONF_TEMPLATE
+
+    @staticmethod
+    def list() -> list[str]:
+        """Return a list of the values."""
+
+        return [c.value for c in ObservationTypes]
+
+
+class ConfigFlowSteps(StrEnum):
```

Comment:
I would prefer to see this as constants.

Reply:
Done in [b7ea3ec](https://github.com/home-assistant/core/pull/122552/commits/b7ea3ec49eade08d7a107bad05f65688e3c5a7c6)